src/components/ActivityFeed/ActivityFeedItem.js

```diff
@@ -0,0 +1,530 @@
+    };
+  }
+
+  getSocialEventData = (event: Object): ?EventData => {
+    const { onRejectInvitation, onAcceptInvitation } = this.props;
+    const { type, username, profileImage } = event;
+
+    let status;
+    if (type === TYPE_ACCEPTED) {
+      status = STATUSES.CONNECTED;
+    } else if (type === TYPE_SENT) {
+      status = STATUSES.REQUESTED;
+    } else {
+      status = null;
+    }
+
+    const data: EventData = {
+      name: username,
+      status,
+      avatarUrl: profileImage,
+      eventData: { ...event },
+      statusAsButton: type === TYPE_SENT,
+    };
+
+    if (type === TYPE_RECEIVED) {
+      data.information = 'Connection request';
+      data.rejectInvitation = () => createAlert(TYPE_REJECTED, event, () => onRejectInvitation(event));
+      data.acceptInvitation = () => onAcceptInvitation(event);
+    }
+
+    return data;
+  }
+
+  getEventData = (event: Object): ?EventData => {
+    switch (event.type) {
+      case USER_EVENT:
+        return this.getUserEventData(event);
+      case TRANSACTION_EVENT:
+        return this.getTransactionEventData(event);
+      case COLLECTIBLE_TRANSACTION:
+        return this.getCollectibleTransactionEventData(event);
+      case BADGE_REWARD_EVENT:
+        return this.getBadgeRewardEventData(event);
+      case TYPE_SENT:
+      case TYPE_RECEIVED:
+      case TYPE_ACCEPTED:
+        return this.getSocialEventData(event);
+      default:
+        return null;
+    }
+  }
+
+  renderSettlement = (event: Object) => {
+    const {
+      assets, supportedAssets, asset, feedType, contacts, contactsSmartAddresses, selectEvent,
+    } = this.props;
+    const { type } = event;
+    const trxData = {
+      hideAmount: true,
+      hideSender: true,
+      txType: 'PLR Network settle',
+    };
+
+    const address = this.isReceived(event) ? event.from : event.to;
+    const contact = findMatchingContact(address, contacts, contactsSmartAddresses) || {};
+    const { decimals = 18 } = getAssetData(assets, supportedAssets, event.asset);
+    const value = formatUnits(event.value, decimals);
+
+    return (
+      <SettlementItem
+        settleData={event.extra}
+        onPress={() => selectEvent({
+          ...event,
+          value,
+          contact,
+          ...trxData,
+        }, type, event)}
+        type={feedType}
+        asset={asset}
+        isPending={event === TX_PENDING_STATUS}
+        supportedAssets={supportedAssets}
+        accountAssets={assets}
+      />
+    );
+  }
+
+  getColor = (color: ?string): ?string => {
+    if (!color) return null;
+    const { theme } = this.props;
+    const colors = getThemeColors(theme);
+    return colors[color] || color;
+  }
+
+  render() {
+    const { event, selectEvent } = this.props;
+
+    if (event.type === TRANSACTION_EVENT && event.tag === PAYMENT_NETWORK_TX_SETTLEMENT) {
+      return this.renderSettlement(event);
+    }
+
+    const itemData = this.getEventData(event);
+
+    if (!itemData) return null;
```

Comment:
This might cause us some problems as we have a sectioned list and we are still returning something. So this "null" might get "visible" empty section with header will be rendered.
I caught a case (but I believe it's happening because we have an issue with connections), when this happens:
A and B users. B user invites A. A rejects it. A invites B. (It is now registered as "connectionDisconnectedEvent" and not as invitation - it must be an error coming from connections). And if user has no more event on that day, empty section is being rendered for that day.
Maybe we can first filter all events by the type that we should show?

Reply:
Can't reproduce that, but added filtering anyway